elastic/beats
# [libbeat] Fix / clarify the module reload logic

## What does this PR do?

The loop logic for the module reloader had a bug that would reload module configurations even if they hadn't changed. We suspect that in some cases this could cause unbounded memory consumption (we are still investigating the exact scope of the problem). This PR fixes the error and restructures the surrounding code to clarify the intended logic.

## Checklist


## How to test this PR locally

With debug output on, any beat with module reloading enabled should output `Number of module configs found: ...` every time it reloads the configuration. Running a beat without this PR should show that this happens at _every_ reload interval, whereas with it it should only reload when a file has changed.
